src/pages/on-chain-governance/_components/ConfirmVoteDialog.tsx

```diff
@@ -171,166 +174,267 @@ export function ConfirmVoteDialog({
               />
-            </>
+            </button>
           )}
         </div>
-        {masterNodeErrorMsg !== "" && (
-          <div className="text-red-600 text-xs mt-1">{masterNodeErrorMsg}</div>
-        )}
       </div>
-      <div className="flex flex-row space-x-2">
-        <button
-          onClick={() => {
-            onClose();
-          }}
-          type="button"
-          className="text-sm w-1/2 py-4 border rounded-sm font-medium border-gray-300 text-primary-500 hover:border-primary-200"
+
+      {masterNodeErrorMsg !== "" ? (
+        <div
+          data-testid="OnChainGovernance.VotingFlow.VoteForProposal.MasternodeErrorMsg"
+          className="text-red-500 text-xs mt-2 md:mb-4 mb-6"
         >
-          CANCEL
-        </button>
+          {masterNodeErrorMsg}
+        </div>
+      ) : (
         <button
           type="button"
-          disabled={isMasterNodeEditClicked || masterNodeID === ""}
-          onClick={() => setIsMasterNodeConfirmClicked(true)}
-          className="text-sm w-1/2 py-4 rounded-sm font-medium border border-primary-50 text-primary-500 bg-primary-50 hover:bg-primary-100 hover:border-primary-100 disabled:bg-gray-50 disabled:border-0 disabled:text-gray-300"
+          onClick={() => {
+            if (rememberMasternodeId === RememberMasterNodeId.Yes) {
+              setRememberMasternodeId(RememberMasterNodeId.No);
+            } else {
+              setRememberMasternodeId(RememberMasterNodeId.Yes);
+            }
+          }}
+          className={classNames(
+            "flex flex-row gap-x-[6px] items-center mt-2 md:mb-4 mb-6 accent-blue-600"
+          )}
         >
-          CONFIRM
+          <input
+            data-testid="OnChainGovernance.VotingFlow.VoteForProposal.MasternodeCheckBox"
+            type="checkbox"
+            onChange={(value) => {
+              if (value.target.checked) {
+                setRememberMasternodeId(RememberMasterNodeId.Yes);
+              } else {
+                setRememberMasternodeId(RememberMasterNodeId.No);
+              }
+            }}
+            checked={rememberMasternodeId === RememberMasterNodeId.Yes}
+          />
+          <div className="text-gray-600 text-sm tracking-[0.0025em]">
+            Remember Masternode ID
+          </div>
         </button>
-      </div>
+      )}
+
+      <UserVote
+        masternodeId={masternodeId}
+        masterNodeErrorMsg={masterNodeErrorMsg}
+        userSelectedVote={userSelectedVote}
+        setUserSelectedVote={setUserSelectedVote}
+      />
+
+      <button
+        data-testid="OnChainGovernance.VotingFlow.VoteForProposal.ContinueButton"
+        type="button"
+        disabled={
+          masterNodeErrorMsg !== "" ||
+          masternodeId === "" ||
+          userSelectedVote === undefined
+        }
+        onClick={() => {
+          const rememberMasternodeObj =
+            getLocalStorageItem("rememberMasternodeId") ?? {};
```

Comment:
i see this line repeated in other places, can you refactor them? onclick callback is getting long, can you move it into a separate function?

Reply:
then maybe add a comment on why do we need to retrieve from localstorage everytime on click

Reply:
updated to shift logic out to function getLocalStorageItem is returning different stuff the one here is getting an object 
getLocalStorageItem("rememberMasternodeId") will get a string/RememberMasternodeId enum